sidechain/top-pool-executor/src/call_operator_mock.rs

```diff
@@ -97,4 +97,17 @@ where
 		remove_call_invoked_lock.push((*shard, calls));
 		Default::default()
 	}
+
+	fn on_block_imported(&self, block: &SignedSidechainBlock::Block) -> Vec<ExecutedOperation> {
+		let signed_top_hashes = block.signed_top_hashes();
+		let executed_operations = signed_top_hashes
+			.iter()
+			.map(|hash| {
+				// Only successfully executed operations are included in a block.
+				ExecutedOperation::success(*hash, TrustedOperationOrHash::Hash(*hash), Vec::new())
+			})
+			.collect();
```

Comment:
In case we really need this elaborate implementation in a mock, we should probably extract a common utility method to get the executed operations from a block?

Reply:
I removed the extra logic completely from on_block_imported. It's now only updating the call statuses of the pool.